src/lxc/start.c

```diff
@@ -850,6 +849,12 @@ static int do_start(void *data)
 		INFO("Unshared CLONE_NEWCGROUP.");
 	}
 
+	/* Setup the container, ip, names, utsname, ... */
+	if (lxc_setup(handler)) {
```

Comment:
So moving the `unshare(CLONE_NEWCGROUP)` after the `pivot_root()` syscall is something which I need to test in a POC before merging as I'm not sure if there isn't a possible interaction between the two. I doubt it but security firsts.